docs/md/executor.md

```diff
@@ -0,0 +1,163 @@
+In a typical workload configuration, an `Executor` uses a supported
+service in the cloud to execute workflows.
+
+## User Guide
+You can configure the type of `Executor` used in your workload by changing the
+`executor` attribute of your workload request.
+
+### `Terra` Executor
+You can execute workflows in a Terra workspace using the `Terra` executor.
+
+The `Terra` executor will...
+
+- Coerce available outputs from an upstream `Source` to a data type acceptable
+  for submission creation
+  (i.e. import a snapshot to the `executor` workspace as a reference)
+- Update the method configuration with the coerced object as its root entity
+  type
+- Launch a submission 
+- Periodically update the statuses of eligible workflows to enable a
+  downstream `Sink` to consume their outputs
+  
+A typical `Terra` executor configuration in the workload request looks like:
+```json
+{
+  "name": "Terra",
+  "workspace": "{workspace-namespace}/{workspace-name}",
+  "methodConfiguration": "{method-configuration-namespace}/{method-configuration-name}",
+  "methodConfigurationVersion": 1,
+  "fromSource": "importSnapshot"
+}
+``` 
+
+And a real-life example for a known method configuration:
+```json
+{
+  "name": "Terra",
+  "workspace": "wfl-dev/CDC_Viral_Sequencing",
+  "methodConfiguration": "wfl-dev/sarscov2_illumina_full",
+  "methodConfigurationVersion": 1,
+  "fromSource": "importSnapshot"
+}
+```
+
+![](./assets/terra-method-configuration.png)
+
+The table below summarises the purpose of each attribute in the above request.
+
+| Attribute                    | Description                                                                                       |
+|------------------------------|---------------------------------------------------------------------------------------------------|
+| `name`                       | Selects the `Terra` executor implementation.                                                      |
+| `workspace`                  | Terra Workspace in which to execute workflows.                                                    |
+| `methodConfiguration`        | Method configuration from which to generate submissions.                                          |
+| `methodConfigurationVersion` | Expected version of the method configuration.                                                     |
+| `fromSource`                 | Instruction to coerce an output from an upstream `Source` to a type understood by this `executor`.|
+
+#### `workspace`
+A `{workspace-namespace}/{workspace-name}` string as it appears in the URL path
+in the Terra UI.
+
+Prerequisites:
+
+- The workspace must exist prior to workload creation.
+- `workflow-launcher@firecloud.org` must be a workspace "Writer" in order to
+  import snapshots to the workspace.
+- The workspace must be compatible with any downstream processing stage that
+  consumes its workflows.
+  For example, we cannot sink a workflow's outputs to an outside workspace.
```

Comment:
I put this in because I was pretty sure it was true.

Is it?
If it is, is this something we should verify at workload creation?

Reply:
I don't think it is. We can write outputs to another workspace no problem. The only issue is that files wouldn't be copied to the new workspace so that is technically a bug!
The idea is that the executor care about what's downstream of it.

Reply:
Thanks for clarifying!  I will remove this example.

Is the file copy issue a bug on our side? I can file a ticket.

Reply:
That would be really helpful. Thanks.